apps/synapse-oauth-signin/src/OAuth2Form.tsx

```diff
@@ -286,12 +285,32 @@ export const OAuth2Form = () => {
     }
   }, [redirectURL])
 
+  const isLoadingProfile =
+    !twoFactorAuthErrorResponse &&
+    !error &&
+    profile &&
+    !isPreviousAuthCheckComplete
+  const isLoadingClientInfo =
+    !error && !oauthClientInfo && !oidcRequestDescription
+  const isRedirecting = redirectURL && oauthClientInfo
+  const promptForTwoFactorAuth = !!twoFactorAuthErrorResponse
+
+  const isLoading =
+    !promptForTwoFactorAuth &&
+    (isLoadingProfile || isLoadingClientInfo || isRedirecting)
```

Comment:
Trying to decompose the loading boolean into pieces that are a bit easier to understand

Reply:
Is this the correct logic for isLoadingProfile?  If the profile is defined, should this be isProfileLoaded?

Reply:
Oh, this is really isLoadingPreviousAuthCheck